rust/src/http2/detect.rs

```diff
@@ -676,6 +676,44 @@ fn http2_escape_header(hd: &parser::HTTP2FrameHeaders, i: u32) -> Vec<u8> {
     return vec;
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn rs_http2_tx_get_header_names(
+    tx: &mut HTTP2Transaction, direction: u8, buffer: *mut *const u8, buffer_len: *mut u32,
+) -> u8 {
+    let mut vec = Vec::new();
+    vec.push('\r' as u8);
```

Comment:
can we do this in a single step? Both '\r's but perhaps even combine it with the construction?

Reply:
I understand you want `vec.extend([b'\r', b'\n'.iter().copied());` looking at the beginning of https://doc.rust-lang.org/std/vec/struct.Vec.html

Going for this

Reply:
Changing it in other places as well

Reply:
Clippy these days suggest to use `vec![]` instead of `new` followed by a `push`.